tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi_ble.ino

```diff
@@ -1073,10 +1089,12 @@ int MI32advertismentCallback(BLE_ESP32::ble_advertisment_t *pStruct)
     TasAutoMutex localmutex(&slotmutex, "Mi32AdCB2");
     switch(UUID){
       case 0xfe95: // std MI?
-      case 0xfdcd: // CGD1?
       {
         MI32ParseResponse(ServiceData, ServiceDataLength, addr, RSSI);
       } break;
+      case 0xfdcd: { // CGD1?
+        MI32ParseCGDK2Packet(ServiceData, ServiceDataLength, addr, RSSI);
+      } break;      
```

Comment:
Doesn't that break the compatibility with CDG1 ?

Reply:
I am afraid you are right.

Reply:
Oh yeah, maybe. :(
This should patch it: 

```     
      case 0xfe95: // std MI?
      case 0xfdcd: // CGD1 & CGDK2
      {
        if (ServiceDataLength == 17){ // CGDK2
          MI32ParseCGDK2Packet(ServiceData, ServiceDataLength, addr, RSSI);
        } else {
          MI32ParseResponse(ServiceData, ServiceDataLength, addr, RSSI);
        }
      } break;
```

Reply:
Now the CGD1 device does not work correctly, it constantly changes from CGD1 to CGDK2.